core/lib/types/src/l1/mod.rs

```diff
@@ -265,159 +264,116 @@ impl L1Tx {
+        for item in &req.transaction.reserved[2..] {
             assert_eq!(item, &U256::zero());
         }
-
-        let calldata = transaction.remove(0).into_bytes().unwrap();
-
-        let signature = transaction.remove(0).into_bytes().unwrap();
-        assert_eq!(signature.len(), 0);
-
+        assert!(req.transaction.signature.is_empty());
         // TODO (SMA-1621): check that `reservedDynamic` are constructed correctly.
-        let _factory_deps_hashes = transaction.remove(0).into_array().unwrap();
-        let _paymaster_input = transaction.remove(0).into_bytes().unwrap();
-        let _reserved_dynamic = transaction.remove(0).into_bytes().unwrap();
-
-        // Decoding metadata
-
-        // Finally, decode the factory dependencies
-        let factory_deps = match dec_ev.remove(0) {
-            Token::Array(factory_deps) => factory_deps,
-            _ => unreachable!(),
-        };
-
-        let factory_deps = factory_deps
-            .into_iter()
-            .map(|token| token.into_bytes().unwrap())
-            .collect::<Vec<_>>();
+        assert!(req.transaction.paymaster_input.is_empty());
+        assert!(req.transaction.reserved_dynamic.is_empty());
 
         let common_data = L1TxCommonData {
-            serial_id,
-            canonical_tx_hash,
-            sender,
+            serial_id: PriorityOpId(req.transaction.nonce.try_into().unwrap()),
+            canonical_tx_hash: H256::from_slice(&req.tx_hash),
+            sender: u256_to_account_address(&req.transaction.from),
             layer_2_tip_fee: U256::zero(),
-            to_mint,
-            refund_recipient,
+            to_mint: req.transaction.reserved[0],
+            refund_recipient: u256_to_account_address(&req.transaction.reserved[1]),
             full_fee: U256::zero(),
-            gas_limit,
-            max_fee_per_gas,
-            gas_per_pubdata_limit,
+            gas_limit: req.transaction.gas_limit,
+            max_fee_per_gas: req.transaction.max_fee_per_gas,
+            gas_per_pubdata_limit: req.transaction.gas_per_pubdata_byte_limit,
             op_processing_type: OpProcessingType::Common,
             priority_queue_type: PriorityQueueType::Deque,
             // DEPRECATED.
-            // TODO (PLA-962): start setting it to 0 for all new transactions.
-            eth_block,
+            eth_block: 0,
         };
 
         let execute = Execute {
-            contract_address,
-            calldata: calldata.to_vec(),
-            factory_deps: Some(factory_deps),
-            value: msg_value,
+            contract_address: u256_to_account_address(&req.transaction.to),
+            calldata: req.transaction.data,
+            factory_deps: Some(req.factory_deps),
+            value: req.transaction.value,
         };
         Ok(Self {
             common_data,
             execute,
-            received_timestamp_ms: unix_timestamp_ms(),
+            received_timestamp_ms: 0,
```

Comment:
probably better to indeed a comment that describes that not all the data will be correct in this method

Reply:
Done, added a comment.